core/templates/dev/head/domain/exploration/InteractionObjectFactorySpec.js

```diff
@@ -0,0 +1,310 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview unit tests for Interaction object factory.
+ */
+
+describe('Interaction object factory', function() {
+  var iof = null;
+  var oof = null;
+  var agof = null;
+  var hof = null;
+  var sof = null;
+  var testInteraction = null;
+  var answerGroupsDict = null;
+  var defaultOutcomeDict = null;
+  var solutionDict = null;
+  var hintsDict = null;
+  var interactionDict = null;
+
+  beforeEach(module('oppia'));
+
+  beforeEach(inject(function($injector) {
+    iof = $injector.get('InteractionObjectFactory');
+    oof = $injector.get('OutcomeObjectFactory');
+    agof = $injector.get('AnswerGroupObjectFactory');
+    hof = $injector.get('HintObjectFactory');
+    sof = $injector.get('SolutionObjectFactory');
+    defaultOutcomeDict = {
+      dest: 'dest_default',
+      feedback: {
+        content_id: 'default_outcome',
+        html: ''
+      },
+      labelled_as_correct: false,
+      param_changes: [],
+      refresher_exploration_id: null,
+      missing_prerequisite_skill_id: null
+    };
+    answerGroupsDict = [{
+      rule_specs: [],
+      outcome: {
+        dest: 'dest_1',
+        feedback: {
+          content_id: 'outcome_1',
+          html: ''
+        },
+        labelled_as_correct: false,
+        param_changes: [],
+        refresher_exploration_id: null,
+        missing_prerequisite_skill_id: null
+      },
+      training_data: ['training_data'],
+      tagged_misconception_id: 'tagged_misconception_id'
```

Comment:
are you sure this is a string?

Reply:
Oh missed it again. Sorry. Done. 
Thanks.